invisible_cities/reco/xy_algorithms_c.pyx

```diff
@@ -0,0 +1,162 @@
+                the overall barycenter all the SiPms above threshold.***
+
+
+                ---------
+                    This kwarg has some physical motivation. It exists to try to partially
+                compensate for the problem that the NEW tracking plane is not continuous even though
+                light can be emitted by the EL at any (x,y). When lm_radius < pitch, the search for
+                SiPMs that might contribute pos and charge to a new Cluster is always centered about
+                the position of hottest_sipm. That is, SiPMs within new_lm_radius of
+                hottest_sipm are taken into account by barycenter(). In contrast, when
+                lm_radius = pitch or pitch*sqrt(2) the search for SiPMs contributing to the new
+                cluster can be centered at any (x,y). Consider the case where at a local maximum
+                there are four nearly equally 'hot' SiPMs. new_local_maximum would yield a pos,
+                pos1, between these hot SiPMs. Searching for SiPMs that contribute to this
+                cluster within new_lm_radius of pos1 might be better than searching searching for
+                SiPMs within new_lm_radius of hottest_sipm.
+                    We should be aware that setting lm_radius to some distance greater than pitch,
+                we allow new_local_maximum to assume any (x,y) but we also create the effect that
+                depending on where new_local_maximum is, more or fewer SiPMs will be
+                within new_lm_radius. This effect does not exist when lm_radius = 0
+                    lm_radius can always be set to 0 mm, but setting it to 15 mm (slightly larger
+                than 10mm * sqrt(2)), should not hurt.
+
+    new_lm_radius = radius, find a new cluster by calling barycenter() on pos/qs of SiPMs within
+                    new_lm_radius of new_local_maximum
+    returns
+    c    : a list of Clusters
+
+    Usage Example
+    In order to create each Cluster from a 3x3 block of SiPMs (where the center SiPM has more
+    charge than the others), one would call:
+    corona(pos, qs,
+           Qthr           =  K1 * units.pes,
+           Qlm            =  K2 * units.pes,
+           lm_radius      =  0  * units.mm , # must be 0
+           new_lm_radius  =  15 * units.mm , # must be 10mm*sqrt(2) or some number slightly larger
+           msipm          =  K3)
+    """
+
+    if len(pos) == 0: raise SipmEmptyList
+    if  sum(qs) == 0: raise SipmZeroCharge
+
+    cdef np.ndarray[long, ndim=1] above_threshold
+    above_threshold = np.where(qs >= Qthr)[0]            # Find SiPMs with qs at least Qthr
+    pos, qs = pos[above_threshold], qs[above_threshold]  # Discard SiPMs with qs less than Qthr
+
+    if len(pos) == 0: raise SipmEmptyListAboveQthr
+    if  sum(qs) == 0: raise SipmZeroChargeAboveQthr
+
+    # if lm_radius or new_lm_radius is negative, just call overall barycenter
+    if lm_radius < 0 or new_lm_radius < 0:
```

Comment:
Can I suggest the use of `<=` here or are we interested in using these variables with value `0`?